homeassistant/components/tradfri/base_class.py

```diff
@@ -23,21 +23,19 @@
 from homeassistant.helpers.dispatcher import async_dispatcher_connect
 from homeassistant.helpers.entity import DeviceInfo, Entity
 
-from .const import DOMAIN, SIGNAL_GW
+from .const import DOMAIN, SIGNAL_GW, TIMEOUT_API
 
 _LOGGER = logging.getLogger(__name__)
 
 
-def handle_error(
-    func: Callable[[Command | list[Command]], Any]
-) -> Callable[[str], Any]:
+def handle_error(func: Callable[..., Any]) -> Callable[..., Any]:
```

Comment:
Here's the example how to do this in our code:
https://github.com/home-assistant/core/blob/6a21b241c0c56d29b20c5cb23c4690a436d80b93/homeassistant/components/vlc_telnet/media_player.py#L70

Reply:
done.

Reply:
Check the import of `Callable` in the module I linked.

Reply:
Python 3.9 has added support for subscription syntax to collections.abc.Callable but currently we're still using Python 3.8 for linting.

https://docs.python.org/3.9/library/typing.html#typing.Callable

Reply:
Changing the import to be from typing solves this problem, but I have several times have pylint/mypy tell me (in python3.9) not use typings.

I have presented a solution that works in for all our versions, so I am not sure what the problem is ? Adding a TypeVar is not simplifying a bit of typing, but nothing else.

Reply:
I run python 3.9 and 3.10 for other reasons, and want the change to be future proof, not limited to 3.8.

Reply:
Not using a TypeVar gives a different result which isn't correct.

Side note: Please keep the conversion unresolved until you address the comment according to the review. Let the reviewer resolve if there's a disagreement.

Reply:
I do not get where you have a different result, the runtime code is exactly the same. typeVar is nothing more than an advanced macro that basically do copy/replace.

I do not see why I cannot make a solution that satisfies mypy/pylint for all our python versions and 3.10. 

But we are not going to discuss this, timeout= demands a change in the func typing, so the alternative is to forget this PR.

Reply:
your example import from typing, whereas the recent documentation tells to import from collections.abc:
https://docs.python.org/3/library/typing.html#callable
I try to follow the documentation.

I at least try to do im cleanly, a good example from zwave_js/addon.py (which is “your” project):
```
F = TypeVar("F", bound=Callable[..., Any])  # pylint: disable=invalid-name


async def wrapper(*args, **kwargs):  # type: ignore
```
which in my mind shows how not to do it. I really wonder if mypy will complain here :smile:

But no need to take this further, the objective was to help end-users, not to discuss mypy details.